be/src/runtime/memory/roaring_allocator.h

```diff
@@ -0,0 +1,35 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#pragma once
+
+#include "runtime/memory/mem_hook_allocator.h"
+
+namespace starrocks {
+
+static MemHookAllocator kDefaultRoaringAllocator = MemHookAllocator{};
+inline thread_local Allocator* tls_roaring_allocator = &kDefaultRoaringAllocator;
```

Comment:
Is there a problem with the initialization order of ThreadLocal variables here?

Reply:
tls_roaring_allocator is only used in Roaring's memory management and there is no global or thread local variables in CRoaring library, so  the initialization order will not be affected here

Reply:
it seems we should port the static to variables to .cpp file?
Is it as expect if file A include kDefaultRoaringAllocator. B include kDefaultRoaringAllocator A.default != B.default?

Reply:
good point. I remove roaring_allocator.h and move these variables to roaring_hook directly